resources/assets/js/components/Application/Experience/Experience.tsx

```diff
@@ -706,14 +733,51 @@ export const ExperienceStep: React.FunctionComponent<ExperienceStepProps> = ({
   handleReturn,
 }) => {
   const intl = useIntl();
+  const [hasError, setHasError] = useState(false);
+
+  // Hack fix to focus on error message after it has been added to the DOM.
+  useEffect(() => {
+    if (hasError) {
+      focusOnElement("experience-step-form-error");
+    }
+  }, [hasError]);
```

Comment:
So, why is this necessary when the submit button also calls `focusOnElement`?

Reply:
I see, the hook triggers focus the first time hasError is set to true, because the div isn't rendered yet when the onClick event happens. On subsequent submit attempts, if the error element is still there it can be focused right in the onClick handler.

Reply:
So I was trying to get the alert message to focus after clicking on the save and continue button. However, the focus is triggered (in the `onClick` event method) before the element exists in the DOM. Therefore, I used the `useEffect` hook to trigger the `focusOnElement` method for that first click after react re-renders and sets `hasError` to true.